Backend/Meta-TV2-api/Meta-TV2-BusinessLayer/BusinessRules.cs

```diff
@@ -201,4 +202,71 @@ public class BusinessRules : IBusinessRules
             return false;
         }
     }
+
+    public async Task<string> GetPosts() {
+        try {
+            var posts = await dataAccess.GetPosts();
+            if (!posts.HasValue)
+                return null;
+            return JsonSerializer.Serialize(posts.Value);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public async Task<string> GetPosts(int id) {
+        try {
+            var posts = await dataAccess.GetPosts(id);
+            if (!posts.HasValue)
+                return null;
+            return JsonSerializer.Serialize(posts.Value);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public async Task<(string, string)> GetPostFileType(int id) {
+        try {
+            var post = await dataAccess.GetPostByPostId(id);
+            if (!post.HasValue)
+                return (null, null);
+            if(post.Value.pathType == "Url")
+                return ("Url", post.Value.filePath);
+            return (post.Value.pathType, post.Value.filePath);
+        } catch (Exception e){
+            return (null, null);
+        }
+    }
+
+    public async Task<bool> AddPost(string post, ICustomFormFile file)
+    {
+        try {
+
+            var deserializedPost = JsonSerializer.Deserialize<Posts>(post);
+
+            if( deserializedPost.pathType != "Url" && deserializedPost.pathType != "Video" && deserializedPost.pathType != "Image" && deserializedPost.pathType != "Html") {
+                return false;
+            }
+
+            if (file == null | deserializedPost.pathType == "")
+                return false;
+            if(deserializedPost.pathType == "Url" & deserializedPost.filePath != "") { // Handle URL case

```

Comment:
Why check for filePath = "" when you already made sure it is only allowed to be "Url", "Video", ...